check/check.go

```diff
@@ -110,6 +113,12 @@ func (p *Params) CheckForUpdate(url string, up *update.Update) (*Result, error)
 		p.Version = 1
 	}
 
+	if osVersion, err := osversion.GetSemanticVersion(); err != nil {
```

Comment:
Does this work on all platforms?

Reply:
I think so and the version string format is covered by `TestString` in `osversion` package.